### PR TITLE
Display help when subcommand is unknown

### PR DESCRIPTION
**Pull request by ntwcklng**
#73 

In addition, we check if the user may typed in a typo e.g.
```js
const args = require('../args')

args
.option('port', 'Specify the port', 3000)
.command('import', 'import')
.command('export', 'export')

const flags = args.parse(process.argv)
```
Running with the unknown Subcommand `omport` would output
```bash
❯ node args.js omport

Unknown Subcommand omport
Did you mean import?

  Usage: args.js [options] [command]

  Commands:

    export  export
    help    Display help
    import  import

  Options:

    -h, --help      Output usage information
    -p, --port <n>  Specify the port (defaults to 3000)
    -v, --version   Output the version number


```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -531,9 +533,19 @@ class Args {
...
+      const availableSubcommands = this.details.commands.map(sub => sub.usage);
+      const suggestSubcommand = didYouMean(subCommand, availableSubcommands);
+      console.log(`\nUnknown Subcommand ${this.printMainColor(subCommand)}`);
+
+      if (suggestSubcommand !== null) {
+        console.log(`Did you mean ${this.printMainColor(suggestSubcommand)}?`);
+      }
+    }
+
+    // Show usage information if "help" or "h" ogit dtdtusption was used
```
Comment (leo) [inline index.js]: > ogit dtdtusption
?

File: package.json [hunk 2]
```diff
@@ -51,6 +51,7 @@
   "dependencies": {
     "camelcase": "4.0.0",
     "chalk": "1.1.3",
+    "didyoumean2": "1.3.0",
```
Comment (leo) [inline package.json]: Is there a equivalent package that doesn't have a number in its name?

File: index.js (hunk 1 above)
Comment (ntwcklng) [inline index.js]: 😨  where does that come from 😅

File: package.json (hunk 2 above)
Comment (ntwcklng) [inline package.json]: Maybe this: https://github.com/aceakash/string-similarity
I will give it locally a go

Comment (leo) [inline package.json]: Sweet, thanks! Numbers only belong into the version tag 😄

**Comment by ntwcklng** [comment]
`string-similarity` is pretty nice. We can specify the `rating` when we want to suggest a sub command. 0.5 was a good median for me

**Comment by leo** [comment]
Perfect! Thank you 😊

**Pull request merged**